### PR TITLE
Dataset class

### PR DESCRIPTION
**Pull request by Stoffle**
Implements `__len__` and `__getitem__`, should be enough for now I think

### CODE REVIEW

File: utils/loader.py [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+        self.transform = transform
+        if root_dir is not None:
+            self.root_dir = root_dir
+        else:
+            # Use the default path, assumes repo was cloned alongside a `data` folder
+            self.root_dir = os.path.join(
+                os.path.dirname(os.path.dirname(os.path.dirname(os.path.abspath(__file__)))),
+                "data"
+            )
+            if not os.path.isdir(self.root_dir):
```
Comment (tmke8) [inline utils/loader.py]: I think this is too far indented?

**Comment by olliethomas** [comment]
<del>Does this assume all the images are in one folder?</del>  Also, should we hold onto the information about what the image is of?

Edit: Read the function at the bottom

File: utils/loader.py [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+            transform (callable, optional): Optional transform to be applied on a sample.
+        """
+        self.root_dir = root_dir
+        self.transform = transform
+        if root_dir is not None:
+            self.root_dir = root_dir
+        else:
+            # Use the default path, assumes repo was cloned alongside a `data` folder
+            self.root_dir = os.path.join(
+                os.path.dirname(os.path.dirname(os.path.dirname(os.path.abspath(__file__)))),
```
Comment (tmke8) [inline utils/loader.py:20]: would be shorter with the new object-oriented path library in python 3.5:

```
from pathlib import Path
self.root_dir = Path(__file__).resolve().parent.parent.parent / Path("data")
```

**Review by tmke8 — CHANGES_REQUESTED** [review]

**Comment by tmke8** [comment]
> Also, should we hold onto the information about what the image is of?

I think we said that his is unnecessary because the categories are so different that the network should have no trouble distinguishing them.

File: utils/loader.py (hunk 2 above)
Comment (Stoffle) [reply utils/loader.py:20]: I didn't know that was a thing, I'll redo the paths with that for readability

File: utils/loader.py (hunk 1 above)
Comment (Stoffle) [reply utils/loader.py]: That was actually intentional, but you're right it makes far more sense outside the `else`

**Comment by tmke8** [comment]
Let's merge it like this for now

**Pull request merged**